### ISSUE TITLE
The email configuration for ssl/tls and starttls is inconsistent

### ISSUE THREAD

**Issue by rawroland** [cakephp/docs#3149]
The option `'tls' => true` actually corresponds to the starttls extension. I received an _"SMTP timeout"_ after using this option for  a mail server which actually supports encryption with ssl/tls, without providing the starttls extension. I think the documentation for the [email configuration](http://book.cakephp.org/2.0/en/core-utility-libraries/email.html#configuration) should specify that the **tls** option actually refers to **starttls**.

**Wrong Configuration with led to SMPT timeout error**
```php
$default = [
    'host' => 'mail.example.com',
    'port' => 465,
    'tls' => true,
];
```

**Correct configuration for ssl/tls encryption**
```php
$default = [
    'host' => 'tls://mail.example.com',
    'port' => 465,
    'tls' => false,
];
```
**PS: This was tested with version 2.7.1, but it can possibly also manifest in version 3, since the configurations are similar**

**Comment by markstory** [comment]
If you have some edits in mind, you can use the 'Edit this page' link to make changes. If not, I can take care of this soon.

**Comment by markstory** [comment]
I think this is fixed with #3172.
